packages/account/src/Components/personal-details/__tests__/personal-details.spec.js

```diff
@@ -289,7 +290,9 @@ describe('<PersonalDetails/>', () => {
         const mock_store = mockStore({});
         render(
             <StoreProvider store={mock_store}>
-                <BrowserRouter>{component}</BrowserRouter>
+                <ExchangeRatesProvider>
```

Comment:
Why do we need to inject the provider in many places? isn't it defeat the purpose of having a provider?

Reply:
we are only using it at the top level of each package which is `app.tsx` the other places are in the test files. 
first I tried to have the whole context and provider in `core` package but I faced many errors . that's why I moved it to hooks package.

Reply:
wondering if there are any general rules/plans around using hooks vs using providers vs both for serving the data?

Reply:
(just noticed that approach does not seem to be consistent across different parts of deriv-app)

Reply:
@wojciech-deriv so you mean instead of having the `useExchangeRate` for getting the `exchange_rates`, we directly get them using `useContext()`?

I don't think there's is any general rules in the project. 
for my case I first tried to just use a hook and handle the api call in it but we could not connect it to the component state and in some cases the `subscribe()` was not called. that's why @yashim-deriv suggested to use a context for handling the subscriptions( for now it's only handling `exchange_rates` call but we could make it more general for being used for al subscription calls) 

as far a I checked we are using the third approach ( providers + hook ) in the wallet project.
please check  `APIProvider`, `APIContext`, and `useAPI` in `packages/api`

Reply:
cc @markwylde-deriv